File: server_code/debug/tgs1000/Script/Dynamic/铜人室火炉.pas
{*********************************************************************
创建人：何永安
创建时间：2009.12.2
创建功能：点亮四个火炉，激活怪物
**********************************************************************}

procedure OnTurnOn(uSource, uDest:integer);
var
    t1, t2, t3, t4  :INTEGER;
begin
    t1 := MapFindDynamicobject('禁止区域', '铜人室火炉1', 'dos_Openned');
    t2 := MapFindDynamicobject('禁止区域', '铜人室火炉2', 'dos_Openned');
    t3 := MapFindDynamicobject('禁止区域', '铜人室火炉3', 'dos_Openned');
    t4 := MapFindDynamicobject('禁止区域', '铜人室火炉4', 'dos_Openned');
    t1 := t1 + t2 + t3 + t4;
    saysystem(uSource, '点亮火炉个数：' + IntToStr(t1));
    if (t1 >= 4) then
    begin
        MapIceMonster('禁止区域', '禁地护卫武士', false);
        saysystem(uSource, '禁地护卫武士：闯王陵者死！');
    end;
end;

procedure OnTurnoff(uSource:integer);
begin
    MapIceMonster('禁止区域', '禁地护卫武士', true);
end;

File: projects/tgs_fanj/tgs_反击/Script/lua/Dynamic/铜人室火炉.lua
--打开触发
function OnTurnOn(uSource, uDest)
  local t1 = M_MapFindDynamicobject(76, '铜人室火炉1', 3);
  local t2 = M_MapFindDynamicobject(76, '铜人室火炉2', 3);
  local t3 = M_MapFindDynamicobject(76, '铜人室火炉3', 3);
  local t4 = M_MapFindDynamicobject(76, '铜人室火炉4', 3);
  t1 = t1 + t2 + t3 + t4;
  P_saysystem(uSource, '点亮火炉个数：' .. t1, 0);
  if t1 >= 4 then
    M_MapIceMonster(76, '禁地护卫武士', false);
    M_MapboNotHItMonster(76, '禁地护卫武士', false);
    P_saysystem(uSource, '禁地护卫武士：闯王陵者死！', 0);
  end;
end;

--关闭触发
function OnTurnOff(uSource)
  M_MapIceMonster(76, '禁地护卫武士', true);
  M_MapboNotHItMonster(76, '禁地护卫武士', true);
end;

